src/target/riscv/riscv.c

```diff
@@ -1596,8 +1596,57 @@ int riscv_enumerate_triggers(struct target *target)
 
 		riscv_set_register_on_hart(target, hartid, GDB_REGNO_TSELECT, tselect);
 
-		LOG_DEBUG("[%d] Found %d triggers", hartid, r->trigger_count[hartid]);
+		LOG_INFO("[%d] Found %d triggers", hartid, r->trigger_count[hartid]);
 	}
 
 	return ERROR_OK;
 }
+
+const char *gdb_regno_name(enum gdb_regno regno)
+{
+	static char buf[32];
+
+	switch (regno) {
+		case GDB_REGNO_ZERO:
+			return "zero";
+		case GDB_REGNO_S0:
+			return "s0";
+		case GDB_REGNO_S1:
+			return "s1";
+		case GDB_REGNO_PC:
+			return "pc";
+		case GDB_REGNO_FPR0:
+			return "fpr0";
+		case GDB_REGNO_FPR31:
+			return "fpr31";
+		case GDB_REGNO_CSR0:
+			return "csr0";
+		case GDB_REGNO_TSELECT:
+			return "tselect";
+		case GDB_REGNO_TDATA1:
+			return "tdata1";
+		case GDB_REGNO_TDATA2:
+			return "tdata2";
+		case GDB_REGNO_MISA:
+			return "misa";
+		case GDB_REGNO_DPC:
+			return "dpc";
+		case GDB_REGNO_DCSR:
+			return "dcsr";
+		case GDB_REGNO_DSCRATCH:
+			return "dscratch";
+		case GDB_REGNO_MSTATUS:
+			return "mstatus";
+		case GDB_REGNO_PRIV:
+			return "priv";
+		default:
+			if (regno <= GDB_REGNO_XPR31) {
+				sprintf(buf, "x%d", regno - GDB_REGNO_XPR0);
+			} else if (regno >= GDB_REGNO_CSR0 && regno <= GDB_REGNO_CSR4095) {
```

Comment:
GDB uses %d. It's also more compact. (eg. csr0x700 vs csr1792) But I'm happy to go with 0x if people prefer that. I was just getting tired of manually converting the gdb numbering to CSR numbers.